ugs-core/src/com/willwinder/universalgcodesender/visualizer/GcodeViewParse.java

```diff
@@ -187,38 +187,37 @@ private static GcodeParser getParser(double arcSegmentLength) {
         // For a line segment list ALL arcs must be converted to lines.
         double minArcLength = 0;
         LineSegment ls;
-        PointSegment ps = endSegment;
-        ps.convertToMetric();
+        endSegment.convertToMetric();
```

Comment:
Actually it was the code analysis tool in IntelliJ that identified this one.

This should be the equivalent to the previous code?
```java
        final PointSegment endSegment = new PointSegment(new Position(1, 1, 1, UnitUtils.Units.INCH), 1);
        endSegment.setIsMetric(false); // We'll assume it's given in inches

        PointSegment ps = endSegment;
        ps.convertToMetric();

        System.out.println(endSegment.point().getX() + " " + endSegment.point().getUnits());
        System.out.println(ps.point().getX() + " " + endSegment.point().getUnits());
```
In this case both print outs will be `25.4 MM` because both ```endSegment``` and ```ps``` referencing the same object. 

For the above to work as expected we would need to make a copy of endSegment:
```java
        final PointSegment endSegment = new PointSegment(new Position(1, 1, 1, UnitUtils.Units.INCH), 1);
        endSegment.setIsMetric(false);

        PointSegment ps = new PointSegment(endSegment);
        ps.convertToMetric();

        System.out.println(endSegment.point().getX() + " " + endSegment.point().getUnits());
        System.out.println(ps.point().getX() + " " + endSegment.point().getUnits());
```

Reply:
You're right. Not sure how I missed that, thanks for the explanation I should have known better!